perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/EventDispatcherImpl.java

```diff
@@ -308,9 +308,15 @@ protected MessageBeans resolveMessage(String msg, Integer idOfMessage) throws In
 
 		//Debug information to check parsing of message.
 		MessageBeans beans = new MessageBeansImpl();
+
+		boolean shouldntAddVo = msg.matches("Group:(.*)created in Vo:(.*)");
```

Comment:
I probably wasn't exact in my comment in previous pull-request. I'm not sure this will work here. Since as we do it, you need to construct (ideally static or one instance) ``Pattern`` with specified regex and ``Pattern.DOT_ALL``. Then on runtime get Matcher out of Pattern like this: ``pattern.matcher(inputString)`` and check it by calling ``matcher.find()``.

I know its more code, but it seems more readable to me. We will also hopefully remove regex logic in near future and will map events by their Java type and not string message.

Reply:
I am not sure that I like the way this patch is going. The original problem, I think, is that audit messages with more perun beans trigger handler for both of those beans (see `SimpleDispatchEventCondition.isApplicable()` ). The condition there could be rewritten to return true iff the presentMask contains only the required beans. 

Given that, the Vo bean should be added to condition on "created in Vo:..." and the VO_F case in CreationEventProcessor should be modified to check for the presence of Group bean and skip Vo creation in that case (this change alone would be sufficient to resolve the issue).